x-pack/plugins/security_solution/public/flyout/left/components/analyze_graph.tsx

```diff
@@ -7,16 +7,58 @@
 
 import type { FC } from 'react';
 import React from 'react';
-import { EuiText } from '@elastic/eui';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { useLeftPanelContext } from '../context';
 import { ANALYZER_GRAPH_TEST_ID } from './test_ids';
+import { Resolver } from '../../../resolver/view';
+import { useTimelineDataFilters } from '../../../timelines/containers/use_timeline_data_filters';
+import { ERROR_TITLE, ERROR_MESSAGE } from '../../translations';
+import { ALERTS_ACTIONS } from '../../../common/lib/apm/user_actions';
+import { useStartTransaction } from '../../../common/lib/apm/use_start_transaction';
+// import { useGlobalTime } from '../../../common/containers/use_global_time';
 
 export const ANALYZE_GRAPH_ID = 'analyze_graph';
 
+const ANALYZER = i18n.translate('xpack.securitySolution.flyout.analyzer', {
+  defaultMessage: 'analyzer',
+});
+
 /**
  * Analyzer graph view displayed in the document details expandable flyout left section under the Visualize tab
  */
 export const AnalyzeGraph: FC = () => {
-  return <EuiText data-test-subj={ANALYZER_GRAPH_TEST_ID}>{'Analyzer graph'}</EuiText>;
+  const { eventId } = useLeftPanelContext();
+  const scopeId = 'fly-out';
+  const { from, to, shouldUpdate, selectedPatterns } = useTimelineDataFilters(false); // always false because flyout is not in timeline
+  const { startTransaction } = useStartTransaction();
```

Comment:
@kqualters-elastic saw this line in `x-pack/plugins/security_solution/public/common/components/header_actions/actions.tsx` but I'm not sure if this is relevant in this case?

Reply:
@christineweng don't think we want to just call startTransaction() in a render, as that will call it on every render, and we want to just run it once when a user opens analyzer I think. Generally this seems to be done in a useCallback or something that will only run a small, fixed number of times, i.e. an api request.